idea/vara-eth/frontend/src/features/codes/lib/queries.ts

```diff
@@ -1,56 +1,51 @@
 import { useQuery } from '@tanstack/react-query';
-import { request } from 'graphql-request';
 
 import { EXPLORER_URL } from '@/shared/config';
-import { graphql } from '@/shared/graphql';
+import { fetchWithGuard } from '@/shared/helpers';
 
 export const CODE_STATUS = {
-  VALIDATION_REQUESTED: 'validation_requested',
-  VALIDATION_FAILED: 'validation_failed',
-  VALIDATED: 'validated',
+  VALIDATION_REQUESTED: 'ValidationRequested',
+  VALIDATION_FAILED: 'ValidationFailed',
+  VALIDATED: 'Validated',
 } as const;
 
-const GET_CODES_QUERY = graphql(`
-  query GetCodes($first: Int!, $offset: Int!) {
-    allCodes(first: $first, offset: $offset) {
-      nodes {
-        id
-        status
-      }
-      totalCount
-    }
-  }
-`);
+export type Code = {
+  id: string;
+  status: string;
+  createdAt: string;
+};
+
+type CodesResponse = {
+  data: Code[];
+  total: number;
+  limit: number;
+  offset: number;
+};
 
 export const useGetAllCodesQuery = (page: number, pageSize: number) => {
-  const first = pageSize;
+  const limit = pageSize;
   const offset = (page - 1) * pageSize;
 
   return useQuery({
-    queryKey: ['allCodes', first, offset],
-    queryFn: () => request(EXPLORER_URL, GET_CODES_QUERY, { first, offset }),
-    select: (data) => data.allCodes,
+    queryKey: ['allCodes', limit, offset],
+    queryFn: async () => {
+      const url = new URL(`${EXPLORER_URL}/codes`);
```

Comment:
Don't you want to move this to fetchWithGuard fn since it's the same in all queries except for the endpoint?

Reply:
I don't think we should move this part into fetchWithGuard. In the queryFn we're only setting the URL endpoint, and that’s the only piece that differs across queries. It feels unnecessary to change fetchWithGuard for this case.